interview-questions.md

```diff
@@ -8,36 +8,47 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: This is not a big deal since Cohort will not have any foreign keys. When we generate Student model, each entry/student/row will have Cohort's id as their foreign key (cohort_id), since each student can only belong to one Cohort. Also we need to add belongs_to in our models folder ... Student < ApplicationRecord
 
-Researched answer:
+Researched answer: My researched answer provided nothing new for me to enter here. Unless im reading this incorrectly, only one model is generated at this point. The second model - Student - when generated, will have one additional column - cohort_id - as its foreign key.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: In order to allocate an item, we need its Id as a parameter when using the following: GET #show - since we need to display a specific item. GET #edit - since we need to edit a specific item. PATCH/PUT #update - since we are updating a specific item. DELETE - since we need to delete a specific item.
 
-Researched answer:
+Researched answer:In order to allocate an item, we need its Id as a parameter when using the following: GET #show - since we need to display a specific item. GET #edit - since we need to edit a specific item. PATCH/PUT #update - since we are updating a specific item. DELETE - since we need to delete a specific item.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: 1. Generate model - will generate db model 2. Generate resource - similar to generate model but with additional items created (routes and controllers ) 3. Generate migration - will generate a change to existing db schema
 
-Researched answer:
+Researched answer: 1. Generate model - The simplest and most lightweight is the model generator. It creates ActiveRecord and ApplicationRecord. 2. Generate resource - Will give us a bit more (than model) to start with. It creates ActiveRecord, ApplicationRecord, controller, and routes. 3. Generate migration - migration is a file that will allow you to make changes to the database.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students -method index - for displaying all of the students
 
-action: "POST" location: /students
+action: "POST" location: /students - method create - for creating a form using ERB file
```

Comment:
Take note of the HTTP verbs here.  POST will be creating a student in the database whereas line 33 is calling on the form since this is a GET request.